app/main.py

```diff
@@ -1,13 +1,22 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"
+            }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {"errors": [format_linter_error(i) for i in errors],
+            "path": file_path, "status": "failed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [{"errors": linter_report[i], "path": i, "status": "passed"}
+            if linter_report[i] == []
+            else format_single_linter_file(i, linter_report[i])
+            for i in linter_report]
```

Comment:
So strange code, simplify it if statement is redundant here, just use format_single_linter_file

Reply:
Because of the operator, my status changes
or should i do it in format_single_linter_file